_im-plugin/ism/error-prevention/resolutions.md

```diff
@@ -0,0 +1,232 @@
+
+---
+
+#### Table of contents
+1. TOC
+{:toc}
+
+
+---
+
+## The index is not the write index
+
+To confirm if one index is a write index, run the following request:
+
+```bash
+GET <index>/_alias?pretty
+```
+
+If the response does not contain `"is_write_index"` : true, the index is not a write index. The following example confirms the index is a write index:
+
+```json
+{
+  "<index>" : {
+    "aliases" : {
+      "<index_alias>" : { 
+        "is_write_index" : true
+      }
+    }
+  }
+}
+```
+
+To set as as write index, run the following request:
+
+```bash
+PUT <index>
+{
+  "aliases": {
+    "<index_alias>" : {
+        "is_write_index" : true
+    }
+  }
+}
+```
+
+## The index does not have an alias
+
+If the index does not have an alias, you can add an alias to the index by running the following request:
+
+```bash
+POST _aliases
+{
+  "actions": [
+    {
+      "add": {
+        "index": "<target_index>",
+        "alias": "<index_alias>"
+      }
+    }
+  ]
+}
+```
+
+## Skipping rollover action is true
+
+In the event that skipping a rollover action occurs, run the following request:
+
+```bash
+ GET <target_index>/_settings?pretty
+```
+
+If the response includes:
+
+```json
+{
+  "index": {
+    "opendistro.index_state_management.rollover_skip": true
+  }
+}
+```
+
+You can reset it by running the following request:
+
+```bash
+PUT <target_index>/_settings
+{
+    "index": {
+      "index_state_management.rollover_skip": false
+  }
+}
+```
+
+## This index has already been rolled over successfully
+
+Remove the [rollover policy from the index]({{site.url}}{{site.baseurl}}/im-plugin/ism/api/#remove-policy-from-index) to prevent this error from recurring. 
+
+## The rollover policy misses rollover_alias index setting
+
+Add a `rollover_alias` index setting to the rollover policy to resolve this issue. Run the following request:
+
+```bash
+PUT _index_template/ism_rollover
+{
+  "index_patterns": ["<index_patterns_in_rollover_policy>"],
+  "template": {
+   "settings": {
+    "plugins.index_state_management.rollover_alias": "<rollover_alias>"
+   }
+ }
+}
+```
+
+## Data too large and exceeding the threshold
+
+Check the [JVM information]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/nodes-info/) and increase the heap memory.
+
+## Maximum shards exceeded
+
+The limit of shards per node, or per index, causes this issue to occur. Check if there is a `total_shards_per_node` limit by running the following request:
+
+```bash
+GET /_cluster/settings
+```
+
+If the response contains `total_shards_per_node`, increase the value of it temporally by running the following request:
+
+```bash
+PUT _cluster/settings
+{
+   "transient":{
+      "cluster.routing.allocation.total_shards_per_node":100
+   }
+}
+```
+
+To check if there is limit of shards for an index, run the following request:
+
+```bash
+GET <index>/_settings/index.routing-
+```
+
+If the response contains the setting below: 
```

Comment:
Similarly to my previous comment, could we combine this and the following sentence into something like "If the response contains the setting in the first example, increase its value or set it to `-1` for unlimited [noun], as shown in the second example:"?